display_information/show-callout/src/main/java/com/esri/samples/show_callout/ShowCalloutSample.java

```diff
@@ -59,31 +59,30 @@ public void start(Stage stage) {
       mapView = new MapView();
       mapView.setMap(map);
 
+      // get the map view's callout
+      Callout callout = mapView.getCallout();
+
       // click event to display the callout
       mapView.setOnMouseClicked(e -> {
-        // check that the primary mouse button was clicked and user is not
-        // panning
+
+        // check that the primary mouse button was clicked and user is not panning
         if (e.isStillSincePress() && e.getButton() == MouseButton.PRIMARY) {
+
+          // hide any previous callout
+          callout.dismiss();
```

Comment:
I think this can be omitted since `showCalloutAt` will "dismiss" the callout for you. Dismissing with a right-click would be helpful to show though.

Reply:
Thanks, that's a good shout. Added a dismiss on secondary click.